diff --git a/src/node/cli.ts b/src/node/cli.ts
index ac4ee32f..1242f84f 100644
--- a/src/node/cli.ts
+++ b/src/node/cli.ts
@@ -697,18 +697,15 @@ export function getResolvedPathsFromArgs(args: UserProvidedArgs): string[] {
 /**
  * Helper function to return the default config file.
  *
- * @param {string} password - Password passed in (usually from generatePassword())
  * @returns The default config file:
  *
  * - bind-addr: 127.0.0.1:8080
  * - auth: password
- * - password: <password>
  * - cert: false
  */
-export function defaultConfigFile(password: string): string {
+export function defaultConfigFile(): string {
   return `bind-addr: 127.0.0.1:8080
 auth: password
-password: ${password}
 cert: false
 `
 }
@@ -733,8 +730,7 @@ export async function readConfigFile(configPath?: string): Promise<ConfigArgs> {
   await fs.mkdir(path.dirname(configPath), { recursive: true })
 
   try {
-    const generatedPassword = await generatePassword()
-    await fs.writeFile(configPath, defaultConfigFile(generatedPassword), {
+    await fs.writeFile(configPath, defaultConfigFile(), {
       flag: "wx", // wx means to fail if the path exists.
     })
     logger.info(`Wrote default config file to ${configPath}`)
