diff --git a/src/node/routes/logout.ts b/src/node/routes/logout.ts
--- a/src/node/routes/logout.ts
+++ b/src/node/routes/logout.ts
@@ -9,5 +9,9 @@
   res.clearCookie(req.cookieSessionName, getCookieOptions(req))
 
   const to = sanitizeString(req.query.to) || "/"
-  return redirect(req, res, to, { to: undefined, base: undefined, href: undefined })
+  return redirect(req, res, to, {
+    to: undefined,
+    base: undefined,
+    href: undefined,
+  })
 })
