diff --git a/src/node/socket.ts b/src/node/socket.ts
--- a/src/node/socket.ts
+++ b/src/node/socket.ts
@@ -5,7 +5,7 @@
 import * as tls from "tls"
 import { Emitter } from "../common/emitter"
 import { generateUuid } from "../common/util"
-import { canConnect, paths } from "./util"
+import { canConnect, paths, toNativeSocketPath } from "./util"
 
 /**
  * Provides a way to proxy a TLS socket. Can be used when you need to pass a
@@ -43,7 +43,7 @@
 
     return new Promise((resolve, reject) => {
       const id = generateUuid()
-      const proxy = net.connect(this.proxyPipe)
+      const proxy = net.connect(toNativeSocketPath(this.proxyPipe))
       proxy.once("connect", () => proxy.write(id))
 
       const timeout = setTimeout(() => {
@@ -77,18 +77,21 @@
   private async startProxyServer(): Promise<net.Server> {
     if (!this._proxyServer) {
       this._proxyServer = this.findFreeSocketPath(this.proxyPipe)
-        .then((pipe) => {
+        .then(async (pipe) => {
           this.proxyPipe = pipe
-          return Promise.all([
-            fs.mkdir(path.dirname(this.proxyPipe), { recursive: true }),
-            fs.rm(this.proxyPipe, { force: true, recursive: true }),
-          ])
+          const nativePipe = toNativeSocketPath(this.proxyPipe)
+          if (!nativePipe.startsWith("\\\\.\\pipe\\")) {
+            await Promise.all([
+              fs.mkdir(path.dirname(this.proxyPipe), { recursive: true }),
+              fs.rm(this.proxyPipe, { force: true, recursive: true }),
+            ])
+          }
         })
         .then(() => {
           return new Promise((resolve) => {
             const proxyServer = net.createServer((p) => this.onProxyConnect.emit(p))
             proxyServer.once("listening", () => resolve(proxyServer))
-            proxyServer.listen(this.proxyPipe)
+            proxyServer.listen(toNativeSocketPath(this.proxyPipe))
           })
         })
     }
@@ -97,10 +100,10 @@
 
   public async findFreeSocketPath(basePath: string, maxTries = 100): Promise<string> {
     let i = 0
-    let path = basePath
-    while ((await canConnect(path)) && i < maxTries) {
-      path = `${basePath}-${++i}`
+    let socketPath = basePath
+    while ((await canConnect(toNativeSocketPath(socketPath))) && i < maxTries) {
+      socketPath = `${basePath}-${++i}`
     }
-    return path
+    return toNativeSocketPath(socketPath)
   }
 }
