diff --git a/src/node/http.ts b/src/node/http.ts
--- a/src/node/http.ts
+++ b/src/node/http.ts
@@ -71,7 +71,7 @@
   return content
     .replace(/{{TO}}/g, (typeof req.query.to === "string" && escapeHtml(req.query.to)) || "/")
     .replace(/{{BASE}}/g, serverOptions.base)
-    .replace(/{{CS_STATIC_BASE}}/g, serverOptions.csStaticBase)
+    .replace(/{{COMPOSERY_STATIC_BASE}}/g, serverOptions.csStaticBase)
     .replace("{{OPTIONS}}", () => escapeJSON(serverOptions))
 }
 
@@ -122,6 +122,9 @@
     case AuthType.Password: {
       // The password is stored in the cookie after being hashed.
       const hashedPasswordFromArgs = req.args["hashed-password"]
+      if (!req.args.password && !hashedPasswordFromArgs) {
+        return false
+      }
       const passwordMethod = getPasswordMethod(hashedPasswordFromArgs)
       const isCookieValidArgs: IsCookieValidArgs = {
         passwordMethod,
