diff --git a/src/node/app.ts b/src/node/app.ts
--- a/src/node/app.ts
+++ b/src/node/app.ts
@@ -10,6 +10,7 @@
 import { disposer } from "./http"
 import { isNodeJSErrnoException } from "./util"
 import { EditorSessionManager, makeEditorSessionManagerServer } from "./vscodeSocket"
+import { toNativeSocketPath } from "./util"
 import { handleUpgrade } from "./wsRouter"
 
 type SocketOptions = { socket: string; "socket-mode"?: string }
@@ -31,7 +32,9 @@
 }
 
 export const listen = async (server: http.Server, opts: ListenOptions) => {
-  if (isSocketOpts(opts)) {
+  const nativeSocket = isSocketOpts(opts) ? toNativeSocketPath(opts.socket) : undefined
+  const isNamedPipe = nativeSocket ? nativeSocket.startsWith("\\\\.\\pipe\\") : false
+  if (isSocketOpts(opts) && !isNamedPipe) {
     try {
       await fs.unlink(opts.socket)
     } catch (error: any) {
@@ -47,7 +50,7 @@
       resolve()
     }
     if (isSocketOpts(opts)) {
-      server.listen(opts.socket, onListen)
+      server.listen(nativeSocket, onListen)
     } else {
       // [] is the correct format when using :: but Node errors with them.
       server.listen(opts.port, opts.host.replace(/^\[|\]$/g, ""), onListen)
@@ -56,10 +59,8 @@
 
   // NOTE@jsjoeio: we need to chmod after the server is finished
   // listening. Otherwise, the socket may not have been created yet.
-  if (isSocketOpts(opts)) {
-    if (opts["socket-mode"]) {
-      await fs.chmod(opts.socket, opts["socket-mode"])
-    }
+  if (isSocketOpts(opts) && opts["socket-mode"] && !isNamedPipe) {
+    await fs.chmod(opts.socket, opts["socket-mode"])
   }
 }
 
